src/main/java/org/orecruncher/dsurround/runtime/sets/BiomeVariables.java

```diff
@@ -66,10 +77,10 @@ public String getId() {
     public String getName() {
         return this.name.get();
     }
-
+    //downfall deprecated or something. No longer scoped out of Biome instances
     @Override
     public float getRainfall() {
-        return this.biome.getDownfall();
+        throw new UnsupportedOperationException("Unimplemented method 'getRainfall'");
```

Comment:
This place has issue on runtime. This should be rewritten.

Reply:
Thanks for the heads up. This method is only for diagnostic purposes- It was originally based on getDownfall, a now deprecated(?) method for biomes, but it's no longer public. Will either edit the script in question and get rid of this altogether or just return null values.

Reply:
> Thanks for the heads up. This method is only for diagnostic purposes- It was originally based on getDownfall, a now deprecated(?) method for biomes, but it's no longer public. Will either edit the script in question and get rid of this altogether or just return null values.

It is used not only for diagnostic purposes, but also in biomes.json L215 (Dust for desert like biomes). For now, let this method return 0f value. Later figured out how to fix it

Reply:
Ohh, I don't see change, you remove getRainfall from biomes.json. Anyway, for now, let it return 0f for this.

Reply:
Indeed, I replaced that with a check for "Dry" biomes, which are decided based on precipitation type. Its the same exact functionality, since the previous implementation checked for downfall < .15f. All biomes <.15f are "dry" anyway, so I just made that logic decide on a biome-level. **Yes, this adds more work for when custom/new biomes are added, but its the best I could do with biome.getDownfall being removed**

Reply:
> Indeed, I replaced that with a check for "Dry" biomes, which are decided based on precipitation type. Its the same exact functionality, since the previous implementation checked for downfall < .15f. All biomes <.15f are "dry" anyway, so I just made that logic decide on a biome-level. **Yes, this adds more work for when custom/new biomes are added, but its the best I could do with biome.getDownfall being removed**

Yep, it's great solution for replace removed biome.getDownfall